util/generate-releases.js

```diff
@@ -0,0 +1,423 @@
+import { format as _format } from 'prettier';
+import { fromStream } from 'ssri';
+const {
+  createReadStream,
+  createWriteStream,
+  ensureDir,
+  readFile,
+  readJsonSync,
+  remove,
+  writeFile,
+} = fs;
+const { whiteBright, green, yellow, red, cyanBright } = chalk;
+const { extractFull } = Seven;
+
+// Options
+const exclude = [
+  'ePi/patch',
+  'hebiiro/DarkenWindow',
+  'hebiiro/DragFilter',
+  'hebiiro/OptimizeEditBox',
+  'rigaya/ffmpegOut',
+  'rigaya/NVEnc',
+  'rigaya/QSVEnc',
+  'rigaya/svtAV1guiEx',
+  'rigaya/VCEEnc',
+  'rigaya/x264guiEx',
+  'rigaya/x265guiEx',
+  'suzune/bakusoku',
+  'suzune/WideDialog',
+];
+const packagesXmlPath = 'v2/data/packages.xml';
+const modXmlPath = 'v2/data/mod.xml';
+
+function format(string) {
+  const xml = string
+    .trim()
+    .replace(/<\?xml-model +/, '<?xml-model ') // Fix `<?xml-model   `
+    .replaceAll('&quot;', '"') // Convert `&quot` to `"`
+    .replaceAll(/<(.+?)>\r?\n?\t+([^<>\t]+?)\r?\n?\t+<\/(.+?)>/g, '<$1>$2</$3>') // Adjust line-breaking
+    .replaceAll(/-->\r?\n?\t?<package>/g, '-->\r\n\r\n\t<package>') // Add line break between a comment and `<package>`
+    .replaceAll(/<\/package>\r?\n?\t?</g, '</package>\r\n\r\n\t<') // Add line break between `<package>`
+    .replaceAll(/\r?\n?\t?<\/packages>/g, '</packages>'); // Remove line break before `</package>`
+  return _format(xml, { parser: 'xml', useTabs: true });
+}
+
+function compareVersion(firstVersion, secondVersion) {
+  if (firstVersion === secondVersion) return 0;
+  const isDate1 = firstVersion.match(/^\d{4}\/\d{2}\/\d{2}$/);
+  const isDate2 = secondVersion.match(/^\d{4}\/\d{2}\/\d{2}$/);
+  if (isDate1 !== isDate2) return 0;
+  if (isDate1 && isDate2) {
+    return compareVersions(
+      firstVersion.replaceAll('/', '.'),
+      secondVersion.replaceAll('/', '.')
+    ); // 2022/02/02 -> 2022.02.02
+  }
+
+  const toSemver = (v) =>
+    v
+      .toLowerCase()
+      .replaceAll(' ', '')
+      .replaceAll('/', '.') // 2022/02/02 -> 2022.02.02
+      .replaceAll(/[^\x20-\x7F]/g, 'z') // v2仮修正-> v2zzz
+      .replaceAll('ver.', '')
+      .replaceAll('ver', '')
+      .replaceAll(/^[vr+]/g, '') // v3 -> 3 and r11 -> 11 and +60 -> 60
+      .replaceAll(/[_+,][vr]?/g, '.') // 1_2.0 -> 1.2.0 and 1.0+3 -> 1.0.3 and 1,v3 -> 1.3
+      .replaceAll('(', '-')
+      .replaceAll(')', '') // 1.0(test) -> 1.0-test
+      .replaceAll(/\d[a-z]/g, (m) => m[0] + '-' + m[1]) // 1.0beta -> 1.0-beta
+      .replaceAll(/[a-z]\d/g, (m) => m[0] + '.' + m[1]) // rc2 -> rc.2
+      .replaceAll(/^\d+\.\d+-/g, (m) => m.slice(0, -1) + '.0-') // 1.0-beta -> 1.0.0-beta
+      .replaceAll(/^\d+-/g, (m) => m.slice(0, -1) + '.0.0-'); // 1-beta -> 1.0.0-beta
+
+  try {
+    return compareVersions(toSemver(firstVersion), toSemver(secondVersion));
+  } catch {
+    return 0;
+  }
+}
+
+async function unzip(zipPath, folderName = null) {
+  const outputPath = resolve(
+    'util/temp',
```

Comment:
```suggestion
    dirname(zipPath),
```
Please see line 261.